hibernate-core/src/test/java/org/hibernate/orm/test/query/hql/AggregateFilterClauseTest.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.query.hql;
+
+import java.util.Date;
+
+import org.hibernate.query.Query;
+
+import org.hibernate.testing.orm.domain.StandardDomainModel;
+import org.hibernate.testing.orm.domain.gambit.EntityOfBasics;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * @author Jan Schatteman
+ */
+@ServiceRegistry
+@DomainModel(standardModels = StandardDomainModel.GAMBIT)
+@SessionFactory
+public class AggregateFilterClauseTest {
+
+	@BeforeAll
+	public void prepareData(SessionFactoryScope scope) {
+		scope.inTransaction(
+				em -> {
+					Date now = new Date();
+
+					EntityOfBasics entity1 = new EntityOfBasics();
+					entity1.setId( 1 );
+					entity1.setTheInt( 5 );
+					entity1.setTheDouble( 5.0 );
+					entity1.setTheDate( now );
+					entity1.setTheBoolean( true );
+					em.persist( entity1 );
+
+					EntityOfBasics entity2 = new EntityOfBasics();
+					entity2.setId( 2 );
+					entity2.setTheInt( 6 );
+					entity2.setTheDouble( 6.0 );
+					entity2.setTheBoolean( true );
+					em.persist( entity2 );
+
+					EntityOfBasics entity3 = new EntityOfBasics();
+					entity3.setId( 3 );
+					entity3.setTheInt( 7 );
+					entity3.setTheDouble( 7.0 );
+					entity3.setTheBoolean( false );
+					entity3.setTheDate( new Date(now.getTime() + 200000L) );
+					em.persist( entity3 );
+
+					EntityOfBasics entity4 = new EntityOfBasics();
+					entity4.setId( 4 );
+					entity4.setTheInt( 13 );
+					entity4.setTheDouble( 13.0 );
+					entity4.setTheBoolean( false );
+					entity4.setTheDate( new Date(now.getTime() + 300000L) );
+					em.persist( entity4 );
+
+					EntityOfBasics entity5 = new EntityOfBasics();
+					entity5.setId( 5 );
+					entity5.setTheDouble( 9.0 );
+					entity5.setTheBoolean( false );
+					em.persist( entity5 );
+				}
+		);
+	}
+
+	@AfterAll
+	public void tearDown(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> session.createQuery( "delete from EntityOfBasics" ).executeUpdate()
+		);
+	}
+
+	@Test
+	public void testSimpleSum(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Long expected = 31L;
+					Query q = session.createQuery( "select sum(eob.theInt) from EntityOfBasics eob" );
+					assertEquals( q.getSingleResult(), expected, "expected " + expected + ", got " + q.getSingleResult() );
```

Comment:
I don't know what kind of method you are calling here (I thought there is only a 3 args `assertEquals` method) but what you seem to anticipate is auto-generated anyway. So please switch to using the following pattern in this class.

```
assertEquals( 31L, q.getSingleResult() );
```

Reply:
Not sure I understand the issue; I'm calling the assertEquals that accepts the expected result, the actual result, and a message in case of failing the boolean test to give some more info as to why it failed (w/ respect to "expected 'true', was 'false' the default no-message method gives), respectively.
I did wrongly switch the first 2 params though (corrected), but again, I don't see the problem?

Reply:
My eyes were too tired apparently 😆 I thought the comma you used in the string was an argument separator. Anyway, in case of a failure, you will get exactly the same message that you are concatenating here, so no need for constructing and passing a message.

Reply:
I disagree, the default (no-message) assertEquals only says that true (equality) was expected and false (non-equality) was the actual; which really doesn't say anything, the message provides what the actual expected result is, and what the query returns, so that saves a debugging trip to get this exact information.

Reply:
It's fine

Reply:
I agree.  I prefer more contextual message here

Reply:
With whom ? ;)